CMakeLists.txt

```diff
@@ -48,7 +48,9 @@ option(alpaka_INSTALL_TEST_HEADER "Install headers of the namespace alpaka::test
 
 include(CMakeDependentOption)
 cmake_dependent_option(alpaka_CHECK_HEADERS "Check all alpaka headers as part of the tests whether they can be compiled standalone." OFF BUILD_TESTING OFF)
+cmake_dependent_option(alpaka_CHECK_HEADERS "Check all alpaka headers as part of the benchmarks whether they can be compiled standalone." OFF alpaka_BUILD_BENCHMARKS OFF)
```

Comment:
-  I have added an additional local variable  as` (alpaka_BUILD_BENCHMARKS  OR BUILD_TESTING)`.

But; Benchmarks were locally not compilable and i got the error, 
`/home/yusufo81/projects/alpaka-dir/alpaka/benchmarks/babelstream/src/babelStreamMainTest.cpp:5:10: fatal error: alpaka/alpaka.hpp: No such file or directory
    5 | #include <alpaka/alpaka.hpp>
      |          ^~~~~~~~~~~~~~~~~~~
`

-  to solve this  i had to change **the condition** to call `add_subdirectory("test/") `at line 167. But with this change, setting benchmark cmake variable became the same thing of setting `BUILD_TESTING`. It is a kind of work-around.
`# Only build the tests if alpaka is the top-level project and BUILD_TESTING or alpaka_BUILD_BENCHMARKS is ON
# Note: alpaka_BUILD_BENCHMARKS is added with an OR besides BUILD_TESTING below.
# Because the benchmark which uses catch2 is not compilable locally without adding test directory as below
# or without setting BUILD_TESTING cmake variable to ON. See PR2299.
if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND (BUILD_TESTING OR alpaka_BUILD_BENCHMARKS))
    add_subdirectory("test/")
endif()
`